File: TugasBesar2.py
def enkripsi_caesar(teks, ubah):
    ciphertext = ""
    for c in teks:
        if c.isalpha():
            ubahlah = chr((ord(c.lower()) - ord('a') + ubah) % 26 + ord('a'))
            ciphertext += ubahlah.upper() if c.isupper() else ubahlah
        else:
            ciphertext += c
    return ciphertext

def dekripsi_caesar(ciphertext, ubah):
    return enkripsi_caesar(ciphertext, 26 - ubah)

def main():
    ubah = 3
    kata = input("Masukan Teks Yang Ingin Dienkripsi: ")
    dekripsi_kata = kata
    print("Kata Yang Di Enkripsi:", dekripsi_kata)
    
    enkripsi = enkripsi_caesar(dekripsi_kata, ubah)
    print("Teks Enkripsi:", enkripsi)
    
    dekripsi = dekripsi_caesar(enkripsi, ubah)
    print("Teks Dekripsi:", dekripsi)

if __name__ == "__main__":
    main()

File: TugasBesar2.cpp
#include <iostream>
#include <string>

using namespace std;

string enkripsiCaesar(const string teks, int ubah) {
    string ciphertext = "";
    for (char c : teks) {
        if (isalpha(c)) {
            char ubahlah = (tolower(c) - 'a' + ubah) % 26 + 'a';
            ciphertext += isupper(c) ? toupper(ubahlah) : ubahlah;
        }
        else {
            ciphertext += c;
        }
    }
    return ciphertext;
}

string deskripsiCaesar(const string ciphertext, int ubah) {
    return enkripsiCaesar(ciphertext, 26 - ubah);
}

int main() {
    int ubah = 3;
    string kata;
    cout<< "Masukan Teks Yang Ingin Dienkripsi :";cin>>kata;
    string dekripsiKata = kata, enkripsi, dekripsi;
    cout << "Kata Yang Di Enkripsi : " << dekripsiKata;
    
    enkripsi = enkripsiCaesar(dekripsiKata, ubah);
    cout <<"\nTeks Enkripsi : " << enkripsi;
    
    dekripsi = deskripsiCaesar(enkripsi, ubah); // Corrected the function name and passed ciphertext
    cout <<"\nTeks Dekripsi : " << dekripsi;
    
    return 0;
}
